hudi-common/src/main/java/org/apache/hudi/internal/schema/convert/AvroInternalSchemaConverter.java

```diff
@@ -70,6 +73,75 @@ public static Schema convert(InternalSchema internalSchema, String name) {
     return buildAvroSchemaFromInternalSchema(internalSchema, name);
   }
 
+  public static InternalSchema pruneAvroSchemaToInternalSchema(Schema schema, InternalSchema originSchema) {
+    List<String> pruneNames = collectColNamesFromSchema(schema);
+    return InternalSchemaUtils.pruneInternalSchema(originSchema, pruneNames);
+  }
+
+  /**
+   * Collect all the leaf nodes names.
+   *
+   * @param schema a avro schema.
+   * @return leaf nodes full names.
+   */
+  private static List<String> collectColNamesFromSchema(Schema schema) {
```

Comment:
does it handle null types recursively?

Reply:
How could we have a recursive null type? Only records, unions, arrays, and maps have sub types

Reply:
what I meant is the items of arary can be a union of null type and string type.

Reply:
Yeah, I think there might be a bug

Reply:
actually I don't think so. The only case that would cause problems would be if there was a recursive union like union[null, union[null, int]] but I don't think that's allowed or would ever be used

Reply:
Wrote unit tests for this. Seems fine